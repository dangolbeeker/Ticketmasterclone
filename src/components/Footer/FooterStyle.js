// Packages
import styled from 'styled-components';

// Styles
import { deviceMin, deviceMax } from '../../styles/DeviceStyle';
import * as Variable from '../../styles/VariableStyle';

// Global Style
import * as Global from '../../styles/GlobalStyle';


export const AppWrapper = styled(Global.AppWrapper)`
    margin: 4% auto 0;
    background-color: ${Variable.hoverGray};
`

export const InfoWrapper = styled.div`
    display: block;
    padding: 25px 5px;

    @media screen and ${deviceMin.mobileS} and ${deviceMax.mobileL} {
        padding: 10px 5px;
    }

    @media screen and ${deviceMin.mobileL} and ${deviceMax.tablet} {
        padding: 15px 5px;
    }

    @media screen and ${deviceMin.tablet} and ${deviceMax.laptop} {
        padding: 20px 5px;
    }
    `

    export const InfoSource = styled.div`
    position: absolute;
    right: 0;
    top: 27px;
    padding-right: 20px;
    img {
        width: 30px;

        @media screen and ${deviceMin.mobileS} and ${deviceMax.mobileL} {
            width: 20px;
        }

        @media screen and ${deviceMin.mobileL} and ${deviceMax.tablet} {
            width: 25px;
        }
    }

    @media screen and ${deviceMin.mobileS} and ${deviceMax.mobileL} {
        top: 12px;
    }

    @media screen and ${deviceMin.mobileL} and ${deviceMax.tablet} {
        top: 14px;
    }

    @media screen and ${deviceMin.tablet} and ${deviceMax.laptop} {
        top: 18px;
    }
    `

    export const Name = styled.span`
    a {
        color: ${Variable.gray};
    }
`
export const LinkWrapper = styled.span`

`

export const Link = styled.a.attrs({
    className: "footerInfo",
  })`
    font-size: 20px;
    font-weight: 700;
    text-decoration: none;
    img {
        float: right;
    }

    @media screen and ${deviceMin.mobileS} and ${deviceMax.mobileL} {
        font-size: 12px;
    }

    @media screen and ${deviceMin.mobileL} and ${deviceMax.tablet} {
        font-size: 16px;
    }

    @media screen and ${deviceMin.tablet} and ${deviceMax.laptop} {
        font-size: 18px;
    }
`
