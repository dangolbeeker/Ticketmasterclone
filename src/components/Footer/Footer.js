// Packages
import React from 'react';

// Styles
import * as Style from "./FooterStyle.js";


class Footer extends React.Component {
    render() {
        return (
            <Style.AppWrapper data-test="footerWrapper">
                <Style.InfoWrapper>
                    <Style.Name>
                        <Style.Link className="footerInfo" data-test="footerLink" href="https://github.com/savasozcan" target="_blank">
                            Savaş Özcan
                        </Style.Link>
                    </Style.Name>
                </Style.InfoWrapper>
                <Style.InfoSource className="source">
                    <Style.LinkWrapper>
                        <Style.Link className="footerInfo" data-test="footerLink" href="https://github.com/savasozcan/ticketmaster-events" target="_blank">
                            <img src="/images/source-code.png" />
                        </Style.Link>
                    </Style.LinkWrapper>
                </Style.InfoSource>
            </Style.AppWrapper>
        )
    }
}

export default Footer
