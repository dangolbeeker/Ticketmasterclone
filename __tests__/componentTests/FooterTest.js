// Packages
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

// Components
import Footer from '../../src/components/Footer/Footer';


configure({ adapter: new Adapter() });

const setUp = (props={}) => {
    const component = shallow(<Footer {...props}/>);
    
    return component;
};

const findTestAttr = (component, attr) => {
    const wrapper = component.find(`[data-test='${attr}']`);

    return wrapper;
};

describe('Footer Component', () => {

    let component;
    beforeEach(() => {
        component = setUp();
    })

    it('It should render without errors', () => {
        const wrapper = findTestAttr(component, 'footerWrapper');
        expect(wrapper.length).toBe(1);
    });

    it('Should render a links', () => {
        const wrapper = findTestAttr(component, 'footerLink');
        expect(wrapper.length).toBe(2)
    });

    test('Snaphot testing', () => {
        const wrapper = renderer.create(<Footer />).toJSON();
        expect(wrapper).toMatchSnapshot();
    })
})