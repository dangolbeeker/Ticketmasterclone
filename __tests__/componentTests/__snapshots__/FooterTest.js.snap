// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Snaphot testing 1`] = `
<div
  className="sc-bdVaJa sc-bwzfXH cXNBeh"
  data-test="footerWrapper"
>
  <div
    className="sc-htpNat eMJjyG"
  >
    <span
      className="sc-ifAKCX OxWJl"
    >
      <a
        className="footerInfo sc-bZQynM footerInfo khtFgc"
        data-test="footerLink"
        href="https://github.com/savasozcan"
        target="_blank"
      >
        Savaş Özcan
      </a>
    </span>
  </div>
  <div
    className="source sc-bxivhb jBnsfr"
  >
    <span
      className="sc-EHOje edRFCv"
    >
      <a
        className="footerInfo sc-bZQynM footerInfo khtFgc"
        data-test="footerLink"
        href="https://github.com/savasozcan/ticketmaster-events"
        target="_blank"
      >
        <img
          src="/images/source-code.png"
        />
      </a>
    </span>
  </div>
</div>
`;
